### PR TITLE
Add before_create hook to generate a key for learning goals

### PR DESCRIPTION
**Pull request by bethanyaconnor**
Learning goals need a key to uniquely identify them through the seed and serialize process (ie getting the content from levelbuilder to production). However, there is no particular need for a user facing key; in current designs, not even curriculum writers need to see the key. Therefore, a before create hook that generates a truly random key makes sense to me -- it'll be a unique alphanumeric code that can easily uniquely identify the learning goal within a rubric.

I considered trying to create a human readable key like we do [for resources](https://github.com/code-dot-org/code-dot-org/blob/082df5a8584ab4a7468eb40e79719e3aedbdfac0/dashboard/app/models/resource.rb#L149), but decided the complexity wasn't worth it given we're not surfacing it to any users. Definitely open to pushback there!

### CODE REVIEW

**Review by davidsbailey — APPROVED** [review]
LGTM, though i am curious what happens when seeding. Do we think the callback gets skipped entirely?

**Comment by bethanyaconnor** [comment]
oh oops, forgot to test seeding 🤦‍♀️ I did write tests for seeding when I implemented it and the Drone run is passing but I'll do a bit of manual testing before merging this PR

**Comment by bethanyaconnor** [comment]
Tested seeding and it works fine. But I also looked into this -- the callbacks are skipped entirely (see [explanation](https://github.com/zdennis/activerecord-import#callbacks). For units, we do [call](https://github.com/code-dot-org/code-dot-org/blob/56d6340c1854d7cde3775d4dc6634855449c2fd8/dashboard/lib/services/script_seed.rb#L293C29-L293C29) `run_callbacks` after import but we don't for any other models afaict (including any rubric models).

So tldr this should only get called when a learning goal is created on levelbuilder and never called on seeding.
